emop-maven-plugin/src/main/java/edu/cornell/RppHandlerMojo.java

```diff
@@ -0,0 +1,166 @@
+package edu.cornell;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Sets;
+import edu.cornell.emop.maven.AgentLoader;
+import edu.cornell.emop.util.Util;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "rpp-handler", requiresDirectInvocation = true, requiresDependencyResolution = ResolutionScope.TEST)
+public class RppHandlerMojo extends MonitorMojo {
+
+    Set<String> criticalSpecsSet;
+    Set<String> backgroundSpecsSet;
+
+    File metaInfoDirectory;
+    @Parameter(property = "criticalSpecsFile", defaultValue = "")
+    private String criticalSpecsFile;
+
+    @Parameter(property = "backgroundSpecsFile", defaultValue = "")
+    private String backgroundSpecsFile;
+
+    @Parameter(property = "javamopAgent")
+    private String javamopAgent;
+
+    /**
+     * Reads a file containing specifications (one on each line), and outputs the set contained in the file.
+     * @param specsFilePath The path to the specifications file.
+     * @return Set of specifications.
+     */
+    protected Set<String> parseSpecsFile(String specsFilePath) {
+        try {
+            return new HashSet<>(Files.readAllLines(new File(specsFilePath).toPath()))
+                    .stream()
+                    .filter(spec -> !spec.isEmpty())
+                    .map(spec -> spec.endsWith("MonitorAspect") ? spec :
+                            spec + "MonitorAspect").collect(Collectors.toSet());
+        } catch (IOException ex) {
+            return new HashSet<>();
+        }
+    }
+
+    /**
+     * Deduces what files we will read for critical and background phase specs.
+     * If the user provides either the criticalSpecsFile or backgroundSpecsFile argument, then we will compute our
+     * critical and background specs based off of the user's choice. Otherwise, we will check whether there were
+     * pre-recorded critical and background specs files from previous runs.
```

Comment:
So, just for clarification, we're allowing developers to have the ability to choose backgroundSpecs and we'll leave it up to them in case some specs are put in both places or some specs are left out completely? Something that might be nice for the future is that if a spec is in both files it will only be run in critical to avoid redundancy